loopout.go

```diff
@@ -666,6 +666,28 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
 		return nil, fmt.Errorf("register spend ntfn: %v", err)
 	}
 
+	// Track whether we have successfully pushed the preimage to the server
+	// so that we do not needlessly continue pushing preimages.
+	var preimagePushed bool
+
+	// pushPreimage pushes our preimage to the server directly if we
+	// have already disclosed our preimage in a sweep and we have not yet
+	// successfully pushed a preimage to the server. This trustlessly speeds
+	// up our swap.
+	pushPreimage := func() {
+		if !preimagePushed && s.state == loopdb.StatePreimageRevealed {
```

Comment:
Probably good to discuss with @alexbosworth the preferred timing for disclosure. Before or after the first tx publication.

Reply:
I don't have a strong preference, but I would say after if pressed

Ideally it will be making no difference since we won't depend on the server ACK

Reply:
Will stick with after. 

> Ideally it will be making no difference since we won't depend on the server ACK

Going to have the client lookup the invoice with lnd, so there's no dependence on the server 👍